src/main/java/seedu/address/logic/util/BalanceCalculationUtil.java

```diff
@@ -19,8 +20,36 @@
     /**
      * Returns an updated balance for {@code payer}
      */
-    public static Balance calculatePayerBalance(Amount amount, Person payer, UniquePersonList payees) {
-        int numberOfInvolvedPersons = numberOfInvolvedPersons(payees);
+    public static Balance calculatePayerBalance(String transactionType, Amount amount, Person payer,
+                                                int numberOfInvolvedPersons) {
+        switch (transactionType) {
+        case AddTransactionCommand.COMMAND_WORD:
+            return calculateAddTransactionPayerBalance(amount, payer, numberOfInvolvedPersons);
+        case DeleteTransactionCommand.COMMAND_WORD:
+            return calculateDeleteTransactionPayerBalance(amount, payer, numberOfInvolvedPersons);
+        default:
+            return null;
+        }
+    }
+    /**
+     * Returns an updated balance for {@code payee}
+     */
+    public static Balance calculatePayeeBalance(String transactionType, Amount amount, Person payee,
+                                                int numberOfInvolvedPersons) {
+        switch (transactionType) {
+        case AddTransactionCommand.COMMAND_WORD:
+            return calculateAddTransactionPayeeBalance(amount, payee, numberOfInvolvedPersons);
+        case DeleteTransactionCommand.COMMAND_WORD:
+            return calculateDeleteTransactionPayeeBalance(amount, payee, numberOfInvolvedPersons);
+        default:
+            return null;
+        }
+    }
+    /**
+     * Calculate new payer balance after a new transaction is added
+     */
+    public static Balance calculateAddTransactionPayerBalance(Amount amount, Person payer,
```

Comment:
In my PR on DeptsTable, I made some changes for this class:
We can call calculateTransactionPayer/PayeeDebt. 
I added some methods in Person ( addBalance), in Balance (ofValue(), getInverse() ). you should check it, to make everything clearer.